saleor/warehouse/management.py

```diff
@@ -33,39 +36,54 @@ def allocate_stocks(order_lines_info: Iterable["OrderLineData"], country_code: s
         return
 
     variants = [line_info.variant for line_info in order_lines_info]
+    if None in variants:
+        raise ValidationError(
+            {
+                "variant": ValidationError(
+                    "Missing variant in order line. Cannot allocate the stock",
+                    code=StockErrorCode.INVALID.value,
+                )
+            }
+        )
 
-    stocks = (
+    stocks = list(
         Stock.objects.select_for_update(of=("self",))
```

Comment:
Will `select_for_update` work when you convert the qs to the list?

Reply:
Yes, before my changes select_for_update was triggered too late (in lazy way), to avoid race condition.